lib/tag.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright © 2020 Atomist, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as semver from "semver";
+import { cleanGitBranch } from "./branch";
+
+export interface NextPrereleaseTagArgs {
+	/** Current branch */
+	branch: string;
+	/** Repository default branch */
+	defaultBranch: string;
+	/** Next release version */
+	nextReleaseVersion: string;
+	/** Current tags */
+	tags: string[];
+}
+
+/**
+ * Return the next prerelease semantic version tag.
+ */
+export function nextPrereleaseTag(args: NextPrereleaseTagArgs): string {
```

Comment:
I wonder if we should move this to the `nextTag` method in `@atomist/skill`. Although this might be npm specific.

Reply:
That's what I was initially going to do, but since I was changing the intent of what was going on, focusing only on prereleases, and still using `nextTag` as a fallback, I did it here.